pkg/certificate/manager_test.go

```diff
@@ -342,25 +340,244 @@ func TestHandleMRCEvent(t *testing.T) {
+					ObjectMeta: v1.ObjectMeta{
+						Name: "mrc",
+					},
+					Spec: v1alpha2.MeshRootCertificateSpec{
+						TrustDomain: "foo.bar.com",
+					},
+					Status: v1alpha2.MeshRootCertificateStatus{
+						State: "unknown",
+					},
+				},
+			},
+			expectedSigningIssuer:    &issuer{Issuer: &fakeIssuer{}, ID: "mrc", TrustDomain: "foo.bar.com", CertificateAuthority: pem.RootCertificate("rootCA")},
+			expectedValidatingIssuer: &issuer{Issuer: &fakeIssuer{}, ID: "mrc", TrustDomain: "foo.bar.com", CertificateAuthority: pem.RootCertificate("rootCA")},
+			expectedError:            false,
+		},
+		{
+			name: "add MRC with no status",
+			mrcEvent: MRCEvent{
+				Type: MRCEventAdded,
+				NewMRC: &v1alpha2.MeshRootCertificate{
+					ObjectMeta: v1.ObjectMeta{
+						Name: "mrc",
+					},
+					Spec: v1alpha2.MeshRootCertificateSpec{
+						TrustDomain: "foo.bar.com",
+					},
+				},
+			},
+			expectedError: false,
 		},
 	}
 
-	for _, tt := range testCases {
-		t.Run(tt.name, func(t *testing.T) {
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
 			assert := tassert.New(t)
-			m := &Manager{}
 
-			err := m.handleMRCEvent(tt.mrcClient, tt.mrcEvent)
-			if !tt.wantErr {
+			cm := &Manager{}
+
+			err := cm.handleMRCEvent(&fakeMRCClient{}, tc.mrcEvent)
+			if tc.expectedError {
+				assert.Error(err)
+			} else {
 				assert.NoError(err)
+			}
+
+			if tc.expectedSigningIssuer != nil {
+				assert.NotNil(cm.signingIssuer)
+				assert.Equal(tc.expectedSigningIssuer, cm.signingIssuer)
+			} else {
+				assert.Nil(cm.signingIssuer)
+			}
+
+			if tc.expectedValidatingIssuer != nil {
+				assert.NotNil(cm.validatingIssuer)
+				assert.Equal(tc.expectedValidatingIssuer, cm.validatingIssuer)
 			} else {
+				assert.Nil(cm.validatingIssuer)
+			}
+		})
+	}
+}
+
+func TestHandleMRCStatusUpdate(t *testing.T) {
+	testCases := []struct {
+		name                     string
+		signingIssuer            issuer
+		validatingIssuer         issuer
+		mrcEvent                 MRCEvent
+		expectedSigningIssuer    issuer
+		expectedValidatingIssuer issuer
+		expectedError            bool
+	}{
+		{
+			name:             "update to validatingRollout status",
+			signingIssuer:    issuer{Issuer: &fakeIssuer{}, ID: "oldmrc", TrustDomain: "foo.bar.com", CertificateAuthority: pem.RootCertificate("rootCA")},
+			validatingIssuer: issuer{Issuer: &fakeIssuer{}, ID: "oldmrc", TrustDomain: "foo.bar.com", CertificateAuthority: pem.RootCertificate("rootCA")},
+			mrcEvent: MRCEvent{
+				Type: MRCEventUpdated,
+				NewMRC: &v1alpha2.MeshRootCertificate{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "newmrc",
+						Namespace: "ns",
+					},
+					Spec: v1alpha2.MeshRootCertificateSpec{
+						TrustDomain: "foo.bar.com",
+					},
+					Status: v1alpha2.MeshRootCertificateStatus{
+						State: constants.MRCStateValidatingRollout,
+					},
+				},
+				OldMRC: &v1alpha2.MeshRootCertificate{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "newmrc",
```

Comment:
should this be `oldmrc` or is this just then status of the MRC?

Reply:
In this test I think I should just be using "mrc" for the MRC's name. Good point!